### PR TITLE
fix(login): rewrite IdentifierFirstLogin step to better manage request parameters

### PR DESCRIPTION
**Pull request by leleueri**
Also add the minimal template example for IDENTIFIER_FIRST_LOGIN

fixes gravitee-io/issues#6975

### CODE REVIEW

**Review by tcompiegne — CHANGES_REQUESTED** [review]
I'm not sure about this solution, maybe we have to rewrite this flow more deeply : 

- GET /identifier-first
- POST /identifier-first
  - Redirect to matching IdP
  - or redirect to AM login page

**Review by tcompiegne — CHANGES_REQUESTED** [review]
PR has been force pushed to rewrite the Identifier-first login flow.

Waiting for a new review.

**Review by remisultan — CHANGES_REQUESTED** [review]
Usage is fine

A few minor comments on code

Postman test are breaking:
image

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/identifierfirst/IdentifierFirstLoginEndpoint.java [hunk 1]
```diff
@@ -69,50 +74,115 @@ public IdentifierFirstLoginEndpoint(TemplateEngine templateEngine, Domain domain
...
+            doInternalRedirect(routingContext);
+            return;
+        }
+
+        // no social providers configured, continue
+        if (socialProviders == null) {
+            doInternalRedirect(routingContext);
+            return;
+        }
+
```
[the remark this answers is not in the document]
Comment (tcompiegne) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/identifierfirst/IdentifierFirstLoginEndpoint.java:146]: We do the check just after

```
 final IdentityProvider identityProvider = socialProviders
                .stream()
                .filter(s -> s.getDomainWhitelist() != null && s.getDomainWhitelist().stream().anyMatch(domainName[1]::equals))
                .findFirst()
                .orElse(null);

        // no IdP has matched, continue
        if (identityProvider == null) {
            doInternalRedirect(routingContext);
            return;
        }
```

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/test/java/io/gravitee/am/gateway/handler/root/resources/endpoint/login/LoginEndpointHandlerTest.java [hunk 2]
```diff
@@ -307,7 +241,7 @@ public void shouldNotInvokeLoginEndpoint_noClient() throws Exception {
                     assertEquals(routingContext.get(FORGOT_ACTION_KEY), resolveProxyRequest(routingContext.request(), routingContext.get(CONTEXT_PATH) + "/forgotPassword", queryParams, true));
                     assertEquals(routingContext.get(REGISTER_ACTION_KEY), resolveProxyRequest(routingContext.request(), routingContext.get(CONTEXT_PATH) + "/register", queryParams, true));
                     assertEquals(routingContext.get(WEBAUTHN_ACTION_KEY), resolveProxyRequest(routingContext.request(), routingContext.get(CONTEXT_PATH) + "/webauthn/login", queryParams, true));
-                    assertEquals(routingContext.get(BACK_TO_LOGIN_IDENTIFIER_ACTION_KEY), resolveProxyRequest(routingContext.request(), routingContext.get(CONTEXT_PATH) + "/login/identifier", queryParamsNoUsername, true));
+                    //assertEquals(routingContext.get(BACK_TO_LOGIN_IDENTIFIER_ACTION_KEY), resolveProxyRequest(routingContext.request(), routingContext.get(CONTEXT_PATH) + "/login/identifier", queryParamsNoUsername, true));
```
[the remark this answers is not in the document]
Comment (tcompiegne) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/test/java/io/gravitee/am/gateway/handler/root/resources/endpoint/login/LoginEndpointHandlerTest.java]: FP

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/identifierfirst/IdentifierFirstLoginEndpoint.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (remisultan) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/identifierfirst/IdentifierFirstLoginEndpoint.java:146]: 👍

**Review by remisultan — APPROVED** [review]
LGTM

**Pull request merged**